### PR TITLE
Add App Engine template

### PR DESCRIPTION
**Pull request by sourced-vince**

### CODE REVIEW

File: community/cloud-foundation/templates/app_engine/examples/flexible_app_engine.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+#
+# Replace <FIXME:appsId> with a name of the parent resource to create the
+# version under. For example, "apps/myapp/services/default".
+
+imports:
+  - path: templates/app_engine/app_engine.py
+    name: app_engine.py
+
+resources:
+  - name: test-app-engine-flexible-${RAND}
```
Comment (sourced-praveenc) [inline community/cloud-foundation/templates/app_engine/examples/flexible_app_engine.yaml]: Example files should not have `{RAND}` in them - Please remove in both example files.

File: community/cloud-foundation/templates/app_engine/app_engine.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+def generate_config(context):
+    """ Generate the deployment configuration. """
+
+    resources = []
+    name = context.properties.get('name', context.env['name'])
+
+    resources = [
+        {
+            'name': name,
+            'type': 'appengine.v1.version',
```
Comment (sourced-praveenc) [inline community/cloud-foundation/templates/app_engine/app_engine.py]: I think this template should def. have the ability to create an `app` first and then the `versions` alongside with it
But this template just creates the `version` right .. presuming that the app engine `app` already exists?

We should be creating the `app` first and then the version - app engine `apps` can be created as defined [here](https://cloud.google.com/appengine/docs/admin-api/reference/rest/v1/apps/create)

The heirarchy is app  -> Service -> Version - See this [here](https://cloud.google.com/appengine/docs/standard/java/an-overview-of-app-engine)
image

_p.s: I know this is a bit of a change but I'm sure they'll come back_

File: community/cloud-foundation/templates/app_engine/app_engine.schema [hunk 3]
```diff
@@ -0,0 +1,895 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+info:
+  title: App Engine
+  author: Sourced Group Inc.
+  description: |
+    Supports creation of an App Engine resource.
+
```
Comment (sourced-praveenc) [inline community/cloud-foundation/templates/app_engine/app_engine.schema]: missing `imports` section here

```
imports:
  - path: app_engine.py
```

File: community/cloud-foundation/templates/app_engine/examples/app_engine.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+imports:
+  - path: templates/app_engine/app_engine.py
+    name: app_engine.py
+
+resources:
+  - name: test-app-engine
+    type: app_engine.py
+    properties:
+      id: <FIXME:project_id>
```
[the remark this answers is not in the document]
Comment (sourced-vince) [reply community/cloud-foundation/templates/app_engine/examples/app_engine.yaml]: Done.

**Pull request closed without merging**